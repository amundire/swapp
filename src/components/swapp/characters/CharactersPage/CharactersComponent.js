import React from 'react';
import Character from '../Character/Character';

//todo: load more button

const CharactersPage = ({allPeople, loadMore}) => {

  return (<section id="charactersList ">
    {allPeople.edges.map(({ node }) => <Character id={node.id} image={node.image} name={node.name}/>)}
    <button onClick={loadMore}>Load More</button>
  </section>
  )
}

export default CharactersPage;