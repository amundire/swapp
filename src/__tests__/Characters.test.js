import React from 'react';
import { shallow } from 'enzyme';
import CharactersComponent from '../components/swapp/characters/CharactersPage/CharactersComponent';

describe('<CharactersPage />', () => {
    it('renders without crashing', () => {
        const data = {
            allPeople: {
                pageInfo: {
                    hasNextPage: false,
                    endCursor: 'Y3Vyc29yLnBlb3BsZS4yMA==',
                },
                edges: [{
                    node: {
                        id: 'people.1',
                        name: 'Luke Skywalker',
                        image: 'https://links.gunaxin.com/content/images/post_images/Luke_Skywalker_039_s_Original_039_Star_Wars_039_Lightsaber_Is_Going_Up_For_Auction_1544116268_4528.jpg',
                    },
                }
                ],
            },
        };
        const wrapper = shallow(<CharactersComponent allPeople={data.allPeople} />);
        expect(wrapper).toMatchSnapshot();
    });
});
