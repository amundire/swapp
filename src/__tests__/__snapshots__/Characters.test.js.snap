// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharactersPage /> renders without crashing 1`] = `
<section
  id="charactersPage"
>
  <div
    className="charactersList"
  >
    <Character
      id="people.1"
      image="https://links.gunaxin.com/content/images/post_images/Luke_Skywalker_039_s_Original_039_Star_Wars_039_Lightsaber_Is_Going_Up_For_Auction_1544116268_4528.jpg"
      name="Luke Skywalker"
    />
  </div>
  <button
    className="loadMoreButton"
  >
    Load More
  </button>
</section>
`;
